Feature: Easier Patient Visits

jniles opened the pull request:
This PR addresses the feature request in #3012.

**Design**
The overall design is to make patient admission and discharge much easier to do for the receptionist.  In 1.x, we previously used a separate area on the registration form, but this caused confusion and was limited to looking up patients by their name or BHIMA identifier.  Normally, returning patients provide their hospital number, so this implementation allows a patient to be discovered based on their hospital number.

We also distinguish between "Ambulatory" and "Hospitalized" patients.  The hospitalized patients will need to be discharged, whilst the ambulatory patients could automatically be discharged immediately after admission.  The rationale is that it would be difficult to know if a patient has left the hospital or not.  We may change this behavior with further testing.

This is ready for an initial review.  @mbayopanda could I get one?

jniles committed:
feat(patients): add returning patient search modal

This commit adds a search modal to find returning patients who are
already registered with the institution.  Once a patient is found, they
can be checked in to the hospital with a diagnoses.

jniles committed:
feat(patients): add hospitalized to admission

This commit adds a differentiation between hospitalized and ambulatory
patients, as well as improving the patient lookup and find flow.  A new
column is added to the visit page "hospitalized" to ensure that patient
visits are classified correctly.

TODO:
 1. What to do about new patients?  Can we admit them automatically?
 2. Add a "search" parameter to the patient search for current
 hospitalized patients.
 3. Automatically discharge patients that are ambulatory patients.

Partially addresses #3012.

jniles committed:
chore(patients): rewrite visits API w/ FilterParser

BREAKING CHANGE: removed the "last" flag for patient visits.  You can
replicate its behavior with "limit" and destructuring.

This commit condenses the visits API by rewriting identical code using
FilterParser.

mbayopanda commented on client/src/modules/patients/record/units/visits.modal.html:
```diff
@@ -1,18 +1,43 @@
 <div class="modal-header">
   <ol class="headercrumb">
-    <li class="static"><span translate>FORM.INFO.PATIENTS</span></li>
```
Nice catch 👍

mbayopanda commented on client/src/modules/patients/record/units/visits.service.js:
```diff
@@ -67,21 +70,19 @@ function VisitService($http, util, Modal) {
+        patient : () => patientUuid,
+        isAdmission : () => isAdmission,
+        currentVisit : () => currentVisit,
+      },
```
👍 much better

mbayopanda reviewed (approved):
LGTM 👍 

I have noticed just two things in the ui : 
- 1. After to have define a visit, the modal is not automatically updated with information about the visit, we have to refresh the page in the browser for seeing that the patient is now "Admit in the hospital (En cours)".

- 2. The "Annuler" button could be "Fermer", this last one is more meaningful according the context of the modal.

![patient admission](https://user-images.githubusercontent.com/5445251/44655931-de331380-a9ee-11e8-9475-da8448dbbe3e.png)

sfount committed:
feat(vouchers): voucher-tools component structure

This commit introduces infrastructure for the `voucher-tools` collection
of components. `bh-voucher-tools-reverse` is introduced to reverse any
given generic transaction.

sfount committed:
feat(vouchers): linking reversal tool utilities

This commit correctly implements and links the utilities for the
`bh-voucher-tools-reverse` component, the journal/reverse route was
updated to provide all information on the voucher that is generated
through the reversal process.

It also adds English translation for the afformentioned component.

sfount committed:
chore(vouchers): cleanup voucher tool markup, lint

This commit fully integrates the `bh-voucher-tools-reverse` into the
journal transaction editing modal.
It also
* Updates the sytle to clearly indicate this is a seperate tool
* Badge style
* Remove unused markup

sfount committed:
feat(vouchers): server voucher tools API end point

This commit introduces an API end point for the `correct` voucher tool.
This method is responsible for both reversing any journal transaction
and creating a generic voucher to replace it with corrected values. This
is achieved by calling the server controllers responsible for journal
transaction reversal and voucher creation and ensuring that the results
are cleaned up on failure.

sfount committed:
feat(vouchers): integrate component, service, API

This commit introduces the core files needed for voucher tools on the
client, the `voucher-tools-correct` component, VoucherToolsService and
templates. It also hooks all of the pieces together to pass information
on the currently edited transaction through to the server correct
transaction API end point.

sfount committed:
feat(vouchers): voucher tools correct component

This commit completes the intial API and client integration for the
voucher tools correct component, including integration with
`bh-voucher-tools-badge` and the additional `correcting` state on
`editTransaction` modal.

sfount committed:
chore(vouchers): correct component style standards

This commit updates the voucher tools correct component to fit the
voucher tools standard. It also cleans up controllers and the Server
API
cleanup endpoint to include Entity `reservsed` flag undoing.

sfount committed:
chore(vouchers): integrate editmodal correct style

This commit update the `editTransaction` modal to properly integrate the
`voucher-tools` `correct` component. This tool differs from reverse tool
as it is not self contained, it requires the containing controller to
provide edited information thorugh the `source` binding. This commit
updates the edit modal to allow the user to make 'corrections' to the
transaction before submitting the correct tool with the latest changes.

sfount committed:
chore(vouchers): clean up development structures

This commit cleans up the controller code for voucher tools components
and updates comments and documentation to ensure the PR is up to project
standards.

sfount committed:
chore(vouchers): cleanup and update structure

This commit removes out of date comments and any additional development
code that should not be merged into production.

jniles committed:
fix(patients): update visits in lookup modal

This commit fixes a bug that was preventing the lookup modal from
displaying the patient visits when they are updated.  The bug originated
from a ngIf that killed the bhVisit component multiple times on the
$digest cycle.  It also cleans up all review comments.

jniles committed:
fix(visits): don't rm hospitalized on discharge

This commit fixes a number of bugs on the returning patient lookup
modal:
 1. Discharging a patient no longer sets hospitalized = 0;
 2. The number of visits listed on the bhPatientVisit component can be
 toggled.

jniles committed:
chore: remove patient checkins

This commit removes all references to patient checkins, a feature that
has been deprecated and now completely removed.

jniles committed:
chore(journal): add French i18n translations

This commit adds the French i18n for @sfount's correction tool.

jniles committed:
chore: add SP to migrate.sql

This commit adds the stored procedure for undoing reversals to
migrate.sql.

jeremielodi committed:
feat(modules) department management

jeremielodi committed:
department - integration test

jeremielodi committed:
departemnt - integration test

jniles committed:
test(journal): integration test for reversals

This commit adds an integration test for the journal corrections.  It
also caught a bug in the cash data SQL that led to an unbalanced
transaction.

mbayopanda committed:
CRUD on entity implementation

jniles committed:
test(vouchers): integration test corrections

This commit adds an integration test for the corrections voucher tool.

jeremielodi committed:
clean up

mbayopanda committed:
Fix translations

mbayopanda committed:
Fix mergin conflict

mbayopanda committed:
integration test for entity

mbayopanda committed:
fix mergin conflicts

mbayopanda committed:
integration tests

mbayopanda committed:
Merge branch 'master' of https://github.com/IMA-WorldHealth/bhima-2.X into crud-person

mbayopanda committed:
Integration test for entity types

jeremielodi committed:
feat(report) ohada profit and loss

jniles committed:
feat(fiscal): correctly calculate closing balance

This commit adds a special route for the closing balance which no longer
sums the entire period total before it.  Instead, it simply fetches
period 0 of the next fiscal year.

jeremielodi committed:
feat(report) profit and loss by year

mbayopanda committed:
getting totals from the next fiscal year

mbayopanda committed:
fix corners fiscal year queris

mbayopanda committed:
use of original ohada reference

jniles committed:
feat(balance): add option to include closing txn

This commit adds an option to the balance sheet to include the closing
transaction on the balance sheet.  Currently, the closing balance is
taken from period 0 of the forthcoming year, however, this does not
provide all the details necessary to work on the closing balance.

jniles committed:
chore(fiscal): create period 13 in procedure

This commit adds the ability to create period 13 in the Stored
Procedure.

jniles committed:
feat(fiscal): write closing txn to period 13

This commit writes the closing balance transaction to period 13 of the
current fiscal year for clarity.  If a fiscal year is closed, this
period will contain the balancing transaction.

Closes #3288.

mbayopanda committed:
Use positive values for liabilities

jniles committed:
fix(test): update balance report tests

This commit updates the tests for the balance report after the period 13
changes made.  It also makes sure the fiscal year is closed correctly.

mbayopanda committed:
fix note in balance sheet

mbayopanda committed:
Merge branch 'master' of https://github.com/IMA-WorldHealth/bhima-2.X into ohada-balance-period-zero-next

jniles committed:
fix(journal): edit txn exchange bug

This commit fixes the exchange rate bug in the editing the Journal.

bors[bot] committed:
Merge #3287

3287: Ohada balance sheet enhancement r=lomamech a=mbayopanda

This PR enhance the ohada balance sheet for getting data from the period zero of the next fiscal year.

Co-authored-by: mbayopanda <mbayopanda@gmail.com>

mbayopanda committed:
closing fiscal year message on the result account

mbayopanda committed:
fix linting errors

greenkeeper[bot] committed:
chore(package): update gulp-iife to version 0.4.0

jniles committed:
chore(fiscal): rename period 13 to closing

This commit removes all references to period 13 and renames it "closing
period."   This renders the system flexible to have the fiscal years of
shorter length.

bors[bot] committed:
Merge #3295

3295: feat(fiscal year) : Message on the result account r=jniles a=mbayopanda

This PR add a message if the selected account in the closing fiscal year modal contains value

![image](https://user-images.githubusercontent.com/5445251/47722999-e52b0c00-dc53-11e8-8278-d568dfdbed40.png)


Co-authored-by: mbayopanda <mbayopanda@gmail.com>

bors[bot] committed:
Merge #3296

3296: Update gulp-iife to the latest version 🚀 r=jniles a=greenkeeper[bot]


## The devDependency [gulp-iife](https://github.com/mariusschulz/gulp-iife) was updated from `0.3.0` to `0.4.0`.
This version is **not covered** by your **current version range**.

If you don’t accept this pull request, your project will work just like it did before. However, you might be missing out on a bunch of new features, fixes and/or performance improvements from the dependency update.

---

[Find out more about this release](https://github.com/mariusschulz/gulp-iife).

<details>
  <summary>FAQ and help</summary>

  There is a collection of [frequently asked questions](https://greenkeeper.io/faq.html). If those don’t help, you can always [ask the humans behind Greenkeeper](https://github.com/greenkeeperio/greenkeeper/issues/new).
</details>

---


Your [Greenkeeper](https://greenkeeper.io) bot :palm_tree:



Co-authored-by: greenkeeper[bot] <greenkeeper[bot]@users.noreply.github.com>

bors[bot] committed:
Merge #3293

3293: fix(journal): edit transaction exchange bug r=mbayopanda a=jniles

This commit fixes the exchange rate bug in the editing the Journal.

Fixes #3292.

Co-authored-by: Jonathan Niles <jonathanwniles@gmail.com>

jniles committed:
Ask bors to delete branches after pass

This commit adds the `delete_merged_branches` key to bors.toml.

bors[bot] committed:
Merge #3298

3298: Ask bors to delete branches after pass r=jniles a=jniles

This commit adds the `delete_merged_branches` key to bors.toml.

Co-authored-by: Jonathan Niles <jniles@users.noreply.github.com>

greenkeeper[bot] committed:
fix(package): update snyk to version 1.105.1

bors[bot] committed:
Merge #3299

3299: Update snyk to the latest version 🚀 r=jniles a=greenkeeper[bot]


## The dependency [snyk](https://github.com/snyk/snyk) was updated from `1.104.2` to `1.105.1`.
This version is **not covered** by your **current version range**.

If you don’t accept this pull request, your project will work just like it did before. However, you might be missing out on a bunch of new features, fixes and/or performance improvements from the dependency update.

---

<details>
<summary>Release Notes for v1.105.1</summary>

<h2><a href="https://urls.greenkeeper.io/snyk/snyk.git/compare/v1.105.0...v1.105.1">1.105.1</a> (2018-10-31)</h2>
<h3>Bug Fixes</h3>
<ul>
<li>fix for assumed patch paths (<a href="https://urls.greenkeeper.io/snyk/snyk.git/commit/370c0be">370c0be</a>)</li>
</ul>
</details>

<details>
<summary>Commits</summary>
<p>The new version differs by 11 commits.</p>
<ul>
<li><a href="https://urls.greenkeeper.io/snyk/snyk/commit/3eb66e089cd9eb2a45f889b048f6c8fb797db9e0"><code>3eb66e0</code></a> <code>Merge pull request #257 from snyk/fix/file-path-fix</code></li>
<li><a href="https://urls.greenkeeper.io/snyk/snyk/commit/4d8aa01e1ec69944c49845ea8296c8360ef429c1"><code>4d8aa01</code></a> <code>tests: update how tests execute patching</code></li>
<li><a href="https://urls.greenkeeper.io/snyk/snyk/commit/7f2175970bbd97aa30db6db1f2a133682062f4ef"><code>7f21759</code></a> <code>Merge pull request #254 from snyk/docs/fix-typos</code></li>
<li><a href="https://urls.greenkeeper.io/snyk/snyk/commit/0c0c33e1faa4cec68a0fcdcdee3bbf7d612b87ad"><code>0c0c33e</code></a> <code>Merge pull request #255 from snyk/test/upgrade-sinon</code></li>
<li><a href="https://urls.greenkeeper.io/snyk/snyk/commit/370c0be96059290be6b3400997fe5179e1dd92af"><code>370c0be</code></a> <code>fix: fix for assumed patch paths</code></li>
<li><a href="https://urls.greenkeeper.io/snyk/snyk/commit/ea7e50b3290dc8cd289bcc252831e702e246978a"><code>ea7e50b</code></a> <code>test: upgrade sinon to 4 (last version to support node4)</code></li>
<li><a href="https://urls.greenkeeper.io/snyk/snyk/commit/83d4fc1a4a4e2aba3f69152cdfae16759ca15aa7"><code>83d4fc1</code></a> <code>chore: upgrade sinon</code></li>
<li><a href="https://urls.greenkeeper.io/snyk/snyk/commit/63ca7a376586a49370b0c0dd06984d69ac8c64ac"><code>63ca7a3</code></a> <code>Merge pull request #251 from pavel-github/docs/report-names-docker-section</code></li>
<li><a href="https://urls.greenkeeper.io/snyk/snyk/commit/c98b8b39779410f70027eab31a97dd910994f332"><code>c98b8b3</code></a> <code>Merge pull request #253 from snyk/feat/climessage-rename</code></li>
<li><a href="https://urls.greenkeeper.io/snyk/snyk/commit/d04d675de783b8daa72a3832d24555c2b0e3cbee"><code>d04d675</code></a> <code>feat: Rename cliMessage to userMessage for consistency</code></li>
<li><a href="https://urls.greenkeeper.io/snyk/snyk/commit/e8b8c6d248a4456c2bde8d681d7327577a4b5565"><code>e8b8c6d</code></a> <code>docs: fix names for snyk report files</code></li>
</ul>
<p>See the <a href="https://urls.greenkeeper.io/snyk/snyk/compare/be98a50cbeab919f0f1f72f876b09d38ba658773...3eb66e089cd9eb2a45f889b048f6c8fb797db9e0">full diff</a></p>
</details>

<details>
  <summary>FAQ and help</summary>

  There is a collection of [frequently asked questions](https://greenkeeper.io/faq.html). If those don’t help, you can always [ask the humans behind Greenkeeper](https://github.com/greenkeeperio/greenkeeper/issues/new).
</details>

---


Your [Greenkeeper](https://greenkeeper.io) bot :palm_tree:



Co-authored-by: greenkeeper[bot] <greenkeeper[bot]@users.noreply.github.com>

jniles committed:
refactor(patients): use async/await

This commit migrates the patients controller on the server to use
async/await where possible.  It also updates the patient reference
calculation to pull the values from the entity_map table.  Finally, it
makes sure that the entity_map table is kept up to date with any changes
made to the patient name.

Contributes to #2458.

bors[bot] committed:
Merge #3273

3273: feat(report) ohada profit and loss r=jniles a=jeremielodi

closes #3268

Co-authored-by: jeremielodi <jeremielodi@gmail.com>

greenkeeper[bot] committed:
fix(package): update snyk to version 1.106.0

bors[bot] committed:
Merge #3228

3228: feat(modules) department management r=jniles a=jeremielodi

closes #2900

Co-authored-by: jeremielodi <jeremielodi@gmail.com>

bors[bot] committed:
Merge #3302

3302: Update snyk to the latest version 🚀 r=jniles a=greenkeeper[bot]


## The dependency [snyk](https://github.com/snyk/snyk) was updated from `1.105.1` to `1.106.0`.
This version is **not covered** by your **current version range**.

If you don’t accept this pull request, your project will work just like it did before. However, you might be missing out on a bunch of new features, fixes and/or performance improvements from the dependency update.

---

<details>
<summary>Release Notes for v1.106.0</summary>

<h1><a href="https://urls.greenkeeper.io/snyk/snyk/compare/v1.105.1...v1.106.0">1.106.0</a> (2018-11-01)</h1>
<h3>Features</h3>
<ul>
<li>display suggestion for using dockerfile (<a href="https://urls.greenkeeper.io/snyk/snyk/commit/006a2f6">006a2f6</a>)</li>
</ul>
</details>

<details>
<summary>Commits</summary>
<p>The new version differs by 2 commits.</p>
<ul>
<li><a href="https://urls.greenkeeper.io/snyk/snyk/commit/b7644e0cb9f6e271a6b050044fc578b3e1511e78"><code>b7644e0</code></a> <code>Merge pull request #260 from snyk/feat/display-tip-dockerfile-option</code></li>
<li><a href="https://urls.greenkeeper.io/snyk/snyk/commit/006a2f607f761bf4a6de1c5e0b2eb4f93b75619a"><code>006a2f6</code></a> <code>feat: display suggestion for using dockerfile</code></li>
</ul>
<p>See the <a href="https://urls.greenkeeper.io/snyk/snyk/compare/3eb66e089cd9eb2a45f889b048f6c8fb797db9e0...b7644e0cb9f6e271a6b050044fc578b3e1511e78">full diff</a></p>
</details>

<details>
  <summary>FAQ and help</summary>

  There is a collection of [frequently asked questions](https://greenkeeper.io/faq.html). If those don’t help, you can always [ask the humans behind Greenkeeper](https://github.com/greenkeeperio/greenkeeper/issues/new).
</details>

---


Your [Greenkeeper](https://greenkeeper.io) bot :palm_tree:



Co-authored-by: greenkeeper[bot] <greenkeeper[bot]@users.noreply.github.com>

bors[bot] committed:
Merge #3283

3283: feat(report) profit and loss by year r=jniles a=jeremielodi

closes #3176

Co-authored-by: jeremielodi <jeremielodi@gmail.com>

jniles committed:
chore: update yarn.lock

jniles committed:
ci: bump Travis Node versions to stable and lts

This commit bumps the nodejs versions to the match the latest stable and
lts versions of node.

jniles committed:
chore: add changes to migrate.sql

This commit add the changes to migrate.sql for the fiscal year
procedures.

greenkeeper[bot] committed:
fix(package): update snyk to version 1.106.1

greenkeeper[bot] committed:
chore(package): update lockfile yarn.lock

bors[bot] committed:
Merge #3307

3307: Update snyk to the latest version 🚀 r=jniles a=greenkeeper[bot]


## The dependency [snyk](https://github.com/snyk/snyk) was updated from `1.106.0` to `1.106.1`.
This version is **not covered** by your **current version range**.

If you don’t accept this pull request, your project will work just like it did before. However, you might be missing out on a bunch of new features, fixes and/or performance improvements from the dependency update.

---

<details>
<summary>Release Notes for v1.106.1</summary>

<h2><a href="https://urls.greenkeeper.io/snyk/snyk/compare/v1.106.0...v1.106.1">1.106.1</a> (2018-11-04)</h2>
<h3>Bug Fixes</h3>
<ul>
<li>avoid docker image name@tag being confused with a real package (<a href="https://urls.greenkeeper.io/snyk/snyk/commit/846ebbd">846ebbd</a>)</li>
</ul>
</details>

<details>
<summary>Commits</summary>
<p>The new version differs by 2 commits.</p>
<ul>
<li><a href="https://urls.greenkeeper.io/snyk/snyk/commit/5ac246cc37c9e7face167fc480fcc37bbd355c7f"><code>5ac246c</code></a> <code>Merge pull request #261 from snyk/fix/exclude_image_name_from_scan</code></li>
<li><a href="https://urls.greenkeeper.io/snyk/snyk/commit/846ebbd98ab92104d1f05d45ddeb5d424765305a"><code>846ebbd</code></a> <code>fix: avoid docker image name@tag being confused with a real package</code></li>
</ul>
<p>See the <a href="https://urls.greenkeeper.io/snyk/snyk/compare/b7644e0cb9f6e271a6b050044fc578b3e1511e78...5ac246cc37c9e7face167fc480fcc37bbd355c7f">full diff</a></p>
</details>

<details>
  <summary>FAQ and help</summary>

  There is a collection of [frequently asked questions](https://greenkeeper.io/faq.html). If those don’t help, you can always [ask the humans behind Greenkeeper](https://github.com/greenkeeperio/greenkeeper/issues/new).
</details>

---


Your [Greenkeeper](https://greenkeeper.io) bot :palm_tree:



Co-authored-by: greenkeeper[bot] <greenkeeper[bot]@users.noreply.github.com>

mbayopanda committed:
fix mergin conflicts

jniles committed:
Merge branch 'master' into fix-fiscal-year-closing-balance-query

jeremielodi committed:
feat(account) refresh the grid after importation

greenkeeper[bot] committed:
fix(package): update snyk to version 1.107.0

greenkeeper[bot] committed:
chore(package): update lockfile yarn.lock

jniles committed:
fix(reversal): undo reversal failures

We needed to wrap the ReverseTransaction() logic in a db.transaction()
to ensure that it was cleaning up its values.

jeremielodi committed:
feat(account management) add account type column in the account grid

jniles committed:
feat(editTransaction): move offline to service

This commit moves the offline validation into the TransactionService.
The motivation is to have better separation of concerns and to share the
offline validation between the voucher tools and the main edit
transaction modal.

bors[bot] committed:
Merge #3308

3308: feat(account) refresh the grid after importation r=jniles a=jeremielodi

closes #3291

Co-authored-by: jeremielodi <jeremielodi@gmail.com>

bors[bot] committed:
Merge #3311

3311: feat(account management) add account type column in the account grid r=jniles a=jeremielodi

closes #3279

Co-authored-by: jeremielodi <jeremielodi@gmail.com>

jniles committed:
test(transaction): add tests for offlineValidation

This commit introduces unit tests for the offlineValidation functions of
the TransactionService.

mbayopanda committed:
Add entity concept into bhima

mbayopanda committed:
Merge branch 'master' of https://github.com/IMA-WorldHealth/bhima-2.X into crud-person

bors[bot] committed:
Merge #3309

3309: Update snyk to the latest version 🚀 r=jniles a=greenkeeper[bot]


## The dependency [snyk](https://github.com/snyk/snyk) was updated from `1.106.1` to `1.107.0`.
This version is **not covered** by your **current version range**.

If you don’t accept this pull request, your project will work just like it did before. However, you might be missing out on a bunch of new features, fixes and/or performance improvements from the dependency update.

---

<details>
<summary>Release Notes for v1.107.0</summary>

<h1><a href="https://urls.greenkeeper.io/snyk/snyk/compare/v1.106.1...v1.107.0">1.107.0</a> (2018-11-05)</h1>
<h3>Features</h3>
<ul>
<li>allow git urls in manifest file (<a href="https://urls.greenkeeper.io/snyk/snyk/commit/3e86cfd">3e86cfd</a>)</li>
</ul>
</details>

<details>
<summary>Commits</summary>
<p>The new version differs by 3 commits.</p>
<ul>
<li><a href="https://urls.greenkeeper.io/snyk/snyk/commit/a7013cd3c72fa731e14a6f47f8372222731790af"><code>a7013cd</code></a> <code>Merge pull request #264 from snyk/feat/allow-urls-on-mafinest-file</code></li>
<li><a href="https://urls.greenkeeper.io/snyk/snyk/commit/404dca2cba9c4812b88bbc7d5636e067a03419a6"><code>404dca2</code></a> <code>chore: update snyk-policy</code></li>
<li><a href="https://urls.greenkeeper.io/snyk/snyk/commit/3e86cfd3caf151fb4934d7bd1adfc7f0d3a461dd"><code>3e86cfd</code></a> <code>feat: allow git urls in manifest file</code></li>
</ul>
<p>See the <a href="https://urls.greenkeeper.io/snyk/snyk/compare/5ac246cc37c9e7face167fc480fcc37bbd355c7f...a7013cd3c72fa731e14a6f47f8372222731790af">full diff</a></p>
</details>

<details>
  <summary>FAQ and help</summary>

  There is a collection of [frequently asked questions](https://greenkeeper.io/faq.html). If those don’t help, you can always [ask the humans behind Greenkeeper](https://github.com/greenkeeperio/greenkeeper/issues/new).
</details>

---


Your [Greenkeeper](https://greenkeeper.io) bot :palm_tree:



Co-authored-by: greenkeeper[bot] <greenkeeper[bot]@users.noreply.github.com>

mbayopanda committed:
Merge branch 'master' of https://github.com/IMA-WorldHealth/bhima-2.X into crud-person

mbayopanda committed:
remove on unique key

mbayopanda committed:
remove unique key for travis

mbayopanda committed:
fix ui select for multiple selection

mbayopanda committed:
Add search by movement reference in stock movement registry

mbayopanda committed:
add unique key on varchar(190)

bors[bot] committed:
Merge #3300

3300: refactor(patients): use async/await r=jniles a=jniles

This commit migrates the patients controller on the server to use async/await where possible.  It also updates the patient reference calculation to pull the values from the entity_map table.  Finally, it makes sure that the entity_map table is kept up to date with any changes made to the patient name.

Contributes to #2458.

Co-authored-by: Jonathan Niles <jonathanwniles@gmail.com>

jniles committed:
fix(vouchers): update tests for latest changes

This commit updates the voucher tests to accounts for the latest changes
in the module.

jniles committed:
fix(cashboxes): load currency modal w/ error

Do not crash when an error occurs looking up the cashbox currency.  On
the create state, the `resolve` would error, causing the `$state` change
to abort and never show the modal.  This logic has been moved into the
modal now and the error handled in that context.

Closes #3305.

jniles committed:
deps: bump ui-grid

This commit fixes the ui-grid sorting behavior by bumping the version to
4.6.4.

Closes #3258

bors[bot] committed:
Merge #3317

3317: deps: bump ui-grid r=jniles a=jniles

This commit fixes the ui-grid sorting behavior by bumping the version to 4.6.4.

Closes #3258

Co-authored-by: Jonathan Niles <jonathanwniles@gmail.com>

mbayopanda committed:
Merge branch 'master' of https://github.com/IMA-WorldHealth/bhima-2.X into filter-movement-reference

bors[bot] committed:
Merge #3316

3316: fix(cashboxes): load currency modal despite error r=jniles a=jniles

Do not crash when an error occurs looking up the cashbox currency.  On the create state, the `resolve` would error, causing the `$state` change to abort and never show the modal.  This logic has been moved into the modal now and the error handled in that context.

Closes #3305.

Co-authored-by: Jonathan Niles <jonathanwniles@gmail.com>

mbayopanda committed:
add tests for reference filter in stock movements

jniles committed:
deps: migrate from bower to yarn

BREAKING CHANGE: removes bower from installation process in favor of
yarn.  Bower officially recommends moving to yarn/webpack/npm and so we
want to get ahead of them breaking our builds by going dark.  It also
simplifies the installation process for new contributors.

Closes #2164.

jniles committed:
fix(i18n): complete EN translations

This commit fixes a gulp warning complaining about missing English
translations.

jniles committed:
docs: remove all mention of bower

This commit removes any mention of bower since we no longer need it to
build and install the application.

bors[bot] committed:
Merge #3315

3315: Add search by movement reference in stock movement registry r=jniles a=mbayopanda

This PR fix the issue #3314 

![2018-11-06_12-01-11](https://user-images.githubusercontent.com/5445251/48060274-dac0c300-e1bb-11e8-8bcb-8094172505cd.gif)

Closes #3314.  Closes #3318.

Co-authored-by: mbayopanda <mbayopanda@gmail.com>

mbayopanda committed:
Merge branch 'master' of https://github.com/IMA-WorldHealth/bhima-2.X into crud-person

mbayopanda committed:
Apply review suggestions

mbayopanda committed:
fix remove route

bors[bot] committed:
Merge #3304

3304: Feature: Close Fiscal Years in last period r=jniles a=jniles

This PR introduces a much-debated feature for closing fiscal years by writing all closing transactions (any movements of money that need to take place before the next fiscal year) into a final, virtual period.

**What is a "virtual period"?**
It's a period that exists without a date.  We already have one - period 0 - which contains our opening balances.  This final period will contain any additional writings to move money around before transferring the balance of all accounts into the next fiscal year's period 0.  In a twelve-month year, this period has the number 13.

**Why make a closing period?**
We need a closing period to house transactions that are purely conceptual - such as writing of income/expense into a balance account.  Previously, we did this magically by simply dumping the result of Income - Expense into Period 0 of the next year on a particular account.  By putting this writing in a closing period, we are able to:
 1. Show all movements of the year _including the final closing operation_ without looking at the next fiscal year's period 0.
 2. Show all the movements of the year _excluding the final closing operation_.
 3. Reproduce in fine detail the closing operation(s) that the accountant did to close the fiscal year.  This might be important for auditing.
 4. The sum of all values in fiscal year N is guaranteed to be the values in fiscal year N + 1's opening balance.

Closes #3288

Co-authored-by: Jonathan Niles <jonathanwniles@gmail.com>
Co-authored-by: Jonathan Niles <jniles@users.noreply.github.com>

jeremielodi committed:
feat(fiscal year) notify while loading data from the server

bors[bot] committed:
Merge #3322

3322: feat(fiscal year) notify while loading data from the server r=jniles a=jeremielodi

closes #3256

Co-authored-by: jeremielodi <jeremielodi@gmail.com>

bors[bot] committed:
Merge #3319

3319: deps: migrate from bower to yarn r=mbayopanda a=jniles

BREAKING CHANGE: removes bower from installation process in favor of yarn.  Bower officially recommends moving to yarn/webpack/npm and so we want to get ahead of them breaking our builds by going dark.  It also simplifies the installation process for new contributors.

Closes #2164.

**TODO**
 - [x] Verify that tests still pass.
 - [x] Verify that application functions as expected.
 - [x] Remove references to `bower` from documentation
 
I might also upgrade gulp to v4, since that will also be a breaking change.

Co-authored-by: Jonathan Niles <jonathanwniles@gmail.com>

greenkeeper[bot] committed:
fix(package): update snyk to version 1.108.1

greenkeeper[bot] committed:
chore(package): update lockfile yarn.lock

bors[bot] committed:
Merge #3324

3324: Update snyk to the latest version 🚀 r=jniles a=greenkeeper[bot]


## The dependency [snyk](https://github.com/snyk/snyk) was updated from `1.107.0` to `1.108.1`.
This version is **not covered** by your **current version range**.

If you don’t accept this pull request, your project will work just like it did before. However, you might be missing out on a bunch of new features, fixes and/or performance improvements from the dependency update.

---

<details>
<summary>Release Notes for v1.108.1</summary>

<h2><a href="https://urls.greenkeeper.io/snyk/snyk/compare/v1.108.0...v1.108.1">1.108.1</a> (2018-11-08)</h2>
<h3>Bug Fixes</h3>
<ul>
<li>show friendly message on monitor timeout (<a href="https://urls.greenkeeper.io/snyk/snyk/commit/bdf1b79">bdf1b79</a>)</li>
<li>update snyk-docker-plugin to handle hosts with ports (<a href="https://urls.greenkeeper.io/snyk/snyk/commit/03f50fb">03f50fb</a>)</li>
</ul>
</details>

<details>
<summary>Commits</summary>
<p>The new version differs by 11 commits.</p>
<ul>
<li><a href="https://urls.greenkeeper.io/snyk/snyk/commit/203f9768bd68a84c80c815e453748af00262de11"><code>203f976</code></a> <code>Merge pull request #269 from snyk/fix/update-docker-plugin-for-tags</code></li>
<li><a href="https://urls.greenkeeper.io/snyk/snyk/commit/4dee6043cf039f12a81a1970d19bce5e6935d683"><code>4dee604</code></a> <code>Merge pull request #268 from snyk/fix/monitor-timeout</code></li>
<li><a href="https://urls.greenkeeper.io/snyk/snyk/commit/03f50fbd4ed4225232194851276afa1007e1fb31"><code>03f50fb</code></a> <code>fix: update snyk-docker-plugin to handle hosts with ports</code></li>
<li><a href="https://urls.greenkeeper.io/snyk/snyk/commit/bdf1b7904713bfaf643c4ef7a501eb1397785f4c"><code>bdf1b79</code></a> <code>fix: show friendly message on monitor timeout</code></li>
<li><a href="https://urls.greenkeeper.io/snyk/snyk/commit/c88bdcecad1defa712baf3710dddfab33c124b56"><code>c88bdce</code></a> <code>Merge pull request #267 from snyk/chore/modernize-run-test</code></li>
<li><a href="https://urls.greenkeeper.io/snyk/snyk/commit/083172fe31fefc5100be0965b6248e3a6fcddcf3"><code>083172f</code></a> <code>chore: lib/snyk-test/run-test.{js -&gt; ts}</code></li>
<li><a href="https://urls.greenkeeper.io/snyk/snyk/commit/a4da0c580b4f179cde1e14c66b017af8be51683d"><code>a4da0c5</code></a> <code>chore: async/await in lib/snyk-test/run-test.js</code></li>
<li><a href="https://urls.greenkeeper.io/snyk/snyk/commit/22f52f57a2167e3bb995d6db0e22b669842de163"><code>22f52f5</code></a> <code>Merge pull request #265 from snyk/feat/faster-lockfile-parsing</code></li>
<li><a href="https://urls.greenkeeper.io/snyk/snyk/commit/c1aca6b5f3a1bc64575c2f90eec1b78b0382d680"><code>c1aca6b</code></a> <code>feat: Bump lockfile parser version for perf update</code></li>
<li><a href="https://urls.greenkeeper.io/snyk/snyk/commit/e7472a0b200ca2012a2226aec88fe377010d7e93"><code>e7472a0</code></a> <code>Merge pull request #262 from snyk/fix/path-node-modules-fallback</code></li>
<li><a href="https://urls.greenkeeper.io/snyk/snyk/commit/3362312348b6f4e3d5705553beb0586d55fb4377"><code>3362312</code></a> <code>fix: Fallback to node_modules for initial wizard test</code></li>
</ul>
<p>See the <a href="https://urls.greenkeeper.io/snyk/snyk/compare/a7013cd3c72fa731e14a6f47f8372222731790af...203f9768bd68a84c80c815e453748af00262de11">full diff</a></p>
</details>

<details>
  <summary>FAQ and help</summary>

  There is a collection of [frequently asked questions](https://greenkeeper.io/faq.html). If those don’t help, you can always [ask the humans behind Greenkeeper](https://github.com/greenkeeperio/greenkeeper/issues/new).
</details>

---


Your [Greenkeeper](https://greenkeeper.io) bot :palm_tree:



Co-authored-by: greenkeeper[bot] <greenkeeper[bot]@users.noreply.github.com>

jniles committed:
deps: remove unused dependencies

This commit removes bower and gulp-util from the dependencies since we
no longer use them.

mbayopanda committed:
Merge branch 'master' of https://github.com/IMA-WorldHealth/bhima-2.X into crud-person

mbayopanda committed:
use of bh components

bors[bot] committed:
Merge #3326

3326: deps: remove unused dependencies r=jniles a=jniles

This commit removes `bower` and `gulp-util` from the dependencies since we no longer use them.

Co-authored-by: Jonathan Niles <jonathanwniles@gmail.com>

bors[bot] committed:
Merge #3313

3313: Entity Management Feature r=jniles a=mbayopanda

This PR add a module for generic entity management : 

![2018-11-05_15-19-38](https://user-images.githubusercontent.com/5445251/48003311-6a0f9d00-e10e-11e8-9174-02867311a8b2.gif)



Co-authored-by: mbayopanda <mbayopanda@gmail.com>

bors[bot] committed:
Merge #3206

3206: Add voucher tools to reverse and correct posted records r=jniles a=jniles

Documented thoroughly in discussion #3063 .

Closes #3063.  Closes #3149.

Co-authored-by: Steven Fountain <stevenfountain@imaworldhealth.org>
Co-authored-by: Jonathan Niles <jonathanwniles@gmail.com>
